packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -0,0 +1,222 @@
+    private emptyNode?: ProxyTarget;
+
+    constructor(
+        public readonly forest: IEditableForest,
+        public readonly tree?: ISharedTree,
+    ) {
+        const observer = new SimpleObservingDependent((token?: InvalidationToken, delta?: Delta.Root): void => {
+            if (token?.isSecondaryInvalidation) {
+                // TODO: this is only an example of after change processing, which is not supported yet by ObjectForest/SharedTree.
+                this.handleAfterChange();
+            } else {
+                this.prepareForEdit();
+            }
+        });
+        this.observers.push(observer);
+        forest.registerDependent(observer);
+    }
+
+    public prepareForEdit(): void {
+        for (const target of this.withCursors) {
+            target.prepareForEdit();
+        }
+        assert(this.withCursors.size === 0, 0x3c0 /* prepareForEdit should remove all cursors */);
+    }
+
+    public free(): void {
+        for (const target of this.withCursors) {
+            target.free();
+        }
+        for (const target of this.withAnchors) {
+            target.free();
+        }
+        for (const observer of this.observers) {
+            this.forest.removeDependent(observer);
+        }
+        assert(this.withCursors.size === 0, 0x3c1 /* free should remove all cursors */);
+        assert(this.withAnchors.size === 0, 0x3c2 /* free should remove all anchors */);
+    }
+
+    public get root(): UnwrappedEditableField {
+        const cursor = this.forest.allocateCursor();
+        const destination = this.forest.root(this.forest.rootField);
+        const cursorResult = this.forest.tryMoveCursorTo(destination, cursor);
+        const targets: ProxyTarget[] = [];
+        if (cursorResult === TreeNavigationResult.Ok) {
+            this.emptyNode?.free();
+            this.emptyNode = undefined;
+            do {
+                targets.push(this.createTarget(cursor));
+            } while (cursor.seek(1) === TreeNavigationResult.Ok);
+        }
+        cursor.free();
+        this.forest.anchors.forget(destination);
+        const rootSchema = lookupGlobalFieldSchema(this.forest.schema, rootFieldKey);
+        const fieldKind = getFieldKind(rootSchema);
+        if (targets.length === 0 && fieldKind.multiplicity !== Multiplicity.Sequence) {
+            targets.push(this.createEmptyTarget());
+        }
+        return proxifyField(this, getFieldKind(rootSchema), targets);
+    }
+
+    public createEmptyTarget(): ProxyTarget {
+        if (!this.emptyNode) {
+            this.emptyNode = new ProxyTarget(this);
+        }
+        return this.emptyNode;
+    }
+
+    public createTarget(cursor: ITreeSubscriptionCursor): ProxyTarget {
+        const path = (cursor as unknown as RootedTextCursor).getPath();
+        const nodePath = pathToString(path);
+        if (!this.nodes.has(nodePath)) {
+            this.nodes.set(nodePath, new ProxyTarget(this, cursor));
```

Comment:
how do we ensure that after a move, nodes are under the correct path in this map?

This seems like a and inefficient problematic way reuse targets.

The way anchors are currently exposed we can't quite use them like this (as map keys), but it would be practical to rework them so that would be practical.

I think a better approach might be to cache already generated child nodes on their parent nodes, and throw them away when invalidated.

The fast path (cache hit case) of having to do a path extraction, a path to string conversion and a map lookup and proxy allocation on each child access is not going to perform great compared to checking for a cached node on the parent.